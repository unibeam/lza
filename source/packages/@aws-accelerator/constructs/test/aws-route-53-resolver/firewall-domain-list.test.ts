/**
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

import * as cdk from 'aws-cdk-lib';
import {
  ResolverFirewallDomainList,
  ResolverFirewallDomainListType,
} from '../../lib/aws-route-53-resolver/firewall-domain-list';
import { snapShotTest } from '../snapshot-test';

const testNamePrefix = 'Construct(ResolverFirewallDomainList): ';

const stack = new cdk.Stack();

// Custom domain list
new ResolverFirewallDomainList(stack, 'TestDomainList', {
  name: 'TestDomainList',
  path: __dirname,
  tags: [],
  type: ResolverFirewallDomainListType.CUSTOM,
  kmsKey: new cdk.aws_kms.Key(stack, 'TestDomainListKey', {}),
  logRetentionInDays: 3653,
});

// Managed domain list
new ResolverFirewallDomainList(stack, 'TestManagedDomainList', {
  name: 'TestManagedDomainList',
  type: ResolverFirewallDomainListType.MANAGED,
  kmsKey: new cdk.aws_kms.Key(stack, 'TestManagedDomainListKey', {}),
  logRetentionInDays: 3653,
});

/**
 * DNS firewall domain list construct test
 */
describe('ResolverFirewallDomainList', () => {
  snapShotTest(testNamePrefix, stack);
});
