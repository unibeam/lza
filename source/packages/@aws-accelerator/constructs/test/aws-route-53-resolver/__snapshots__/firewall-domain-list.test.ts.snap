// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResolverFirewallDomainList Construct(ResolverFirewallDomainList):  Snapshot Test 1`] = `
{
  "Resources": {
    "CustomResolverManagedDomainListCustomResourceProviderHandler9F7C9581": {
      "DependsOn": [
        "CustomResolverManagedDomainListCustomResourceProviderRole33DECC65",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "REPLACED-GENERATED-NAME.zip",
        },
        "Handler": "__entrypoint__.handler",
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "CustomResolverManagedDomainListCustomResourceProviderRole33DECC65",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomResolverManagedDomainListCustomResourceProviderLogGroupAA48B1A0": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "KmsKeyId": {
          "Fn::GetAtt": [
            "TestManagedDomainListKeyCB86CE9D",
            "Arn",
          ],
        },
        "LogGroupName": {
          "Fn::Join": [
            "",
            [
              "/aws/lambda/",
              {
                "Ref": "CustomResolverManagedDomainListCustomResourceProviderHandler9F7C9581",
              },
            ],
          ],
        },
        "RetentionInDays": 3653,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Delete",
    },
    "CustomResolverManagedDomainListCustomResourceProviderRole33DECC65": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
        "Policies": [
          {
            "PolicyDocument": {
              "Statement": [
                {
                  "Action": [
                    "route53resolver:ListFirewallDomainLists",
                  ],
                  "Effect": "Allow",
                  "Resource": "*",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "Inline",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "TestDomainList9DC6C806": {
      "Properties": {
        "DomainFileUrl": {
          "Fn::Sub": "REPLACED-GENERATED-NAME.zip",
        },
        "Tags": [
          {
            "Key": "Name",
            "Value": "TestDomainList",
          },
        ],
      },
      "Type": "AWS::Route53Resolver::FirewallDomainList",
    },
    "TestDomainListKey95870FD5": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "KeyPolicy": {
          "Statement": [
            {
              "Action": "kms:*",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
    "TestManagedDomainListE1CDFDDE": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "CustomResolverManagedDomainListCustomResourceProviderLogGroupAA48B1A0",
      ],
      "Properties": {
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomResolverManagedDomainListCustomResourceProviderHandler9F7C9581",
            "Arn",
          ],
        },
        "listName": "TestManagedDomainList",
        "region": {
          "Ref": "AWS::Region",
        },
      },
      "Type": "Custom::ResolverManagedDomainList",
      "UpdateReplacePolicy": "Delete",
    },
    "TestManagedDomainListKeyCB86CE9D": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "KeyPolicy": {
          "Statement": [
            {
              "Action": "kms:*",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;
